### PR TITLE
Adding support for typed errors

### PR DESCRIPTION
**Pull request by evgenykuzyakov**

### CODE REVIEW

File: src.ts/account.ts [hunk 1]
```diff
@@ -88,13 +88,13 @@ export class Account {
             waitTime *= TX_STATUS_RETRY_WAIT_BACKOFF;
             i++;
         }
-        throw new Error(`Exceeded ${TX_STATUS_RETRY_NUMBER} status check attempts for transaction ${base_encode(txHash)}.`);
+        throw new TypedError(`Exceeded ${TX_STATUS_RETRY_NUMBER} status check attempts for transaction ${base_encode(txHash)}.`, 'RetiesExceeded');
```
Comment (vgrichina) [inline src.ts/account.ts]: RetriesExceeded

File: src.ts/account.ts [hunk 2]
```diff
@@ -88,13 +88,13 @@ export class Account {
...
     }
 
     private async signAndSendTransaction(receiverId: string, actions: Action[]): Promise<FinalExecutionOutcome> {
         await this.ready;
         if (!this._accessKey) {
-            throw new Error(`Can not sign transactions, initialize account with available public key in Signer.`);
+            throw new TypedError(`Can not sign transactions, initialize account with available public key in Signer.`, 'CantSignError');
```
Comment (vgrichina) [inline src.ts/account.ts]: KeyNotFound

Error suffix generally feels redundant as well

File: src.ts/account.ts [hunk 3]
```diff
@@ -107,7 +107,7 @@ export class Account {
         try {
             result = await this.connection.provider.sendTransaction(signedTx);
         } catch (error) {
-            if (error.message === '[-32000] Server error: send_tx_commit has timed out.') {
```
Comment (vgrichina) [inline src.ts/account.ts]: looks like should have both checks for compatibility with testnet

Comment (evgenykuzyakov) [reply src.ts/account.ts]: I moved this check to provider to generate TimeoutError there

**Pull request merged**